clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -480,6 +491,7 @@ offline_instru_t::record_instr_encodings(void *drcontext, app_pc tag_pc,
     log_(3, "%s: new block id " UINT64_FORMAT_STRING " for %p\n", __FUNCTION__,
          encoding_id_, tag_pc);
     per_block->id = encoding_id_++;
+    per_block->encoding_length_start = encoding_length_;
```

Comment:
It took me a while (offline discussion) to understand why we didn't simply store one instruction per encoding entry and have the encoding ID be per-instruction: it seems much simpler at first, avoiding all this encoding length tracking and avoiding the raw2trace split code paths and new raw2trace map.  The only downside seems to be the extra space from the 3 64-bit fields per encoding entry which is a lot -- though for small-gencode apps like those just using VDSO it still seems better that way as the encoding entry count is so small.

Please update the PR description to explain *why* this adds this extra complexity: to save space.

Please add comment(s) in the code too: maybe in trace_entry.h and also in raw2trace.h.

Reply:
Added it to the PR description and to trace_entry.h.